[WIP] http4s - migrate from Blaze to Ember

ivan-klass opened the pull request:
Ember seems to be more modern and especially better supported in terms of cross-platform builds

adamw commented:
Oh nice, I was worried you abandoned the PR ;)

ivan-klass commented on doc/tutorials/07_cats_effect.md:
```diff
@@ -154,12 +154,11 @@ object HelloWorldTapir extends IOApp:
-      .use(_ => IO.never)
-      .as(ExitCode.Success)
+      .build
+      .useForever
```
```suggestion
      .useForever
      .as(ExitCode.Success)
```

ivan-klass commented on doc/tutorials/07_cats_effect.md:
```diff
@@ -226,17 +225,16 @@ object HelloWorldTapir extends IOApp:
-      .use(_ => IO.never)
-      .as(ExitCode.Success)
+      .build
+      .useForever
```
```suggestion
      .useForever
      .as(ExitCode.Success)
```

ivan-klass commented:
@adamw when you have a chance, could you please take a look on #4267 so I can rebase this PR on top of that?

adamw commented:
@ivan-klass sure thing, done :)

ivan-klass commented:
To whom it may concern: ember builder doesn't have an option for a specific execution context, so test request handling is not in the global one as it was before. It seems like race conditions are observed when dealing with some web-socket tests.

adamw commented:
@ivan-klass sorry for the delay - the global EC was used before only because it was the easiest one to obtain ;). I don't think there's anything in the tests that would rely on the global one being used: each server should be independent anyway, regardless on the thread pool on which tasks/futures are scheduled. So I would look for causes of the exceptions elsewhere, unfortunately.

ivan-klass committed:
http4s: Migrate from blaze to ember

Replace the blaze server/client backend with ember across the http4s
modules, examples, docs, and perf tests. Includes connection idle
timeout handling, websocket pong/close-frame fixes, multipart decoding
fix, and test interpreter cleanup using ember's default idle timeout.
